lib/puppet/provider/package/yum.rb

```diff
@@ -151,6 +154,36 @@ def self.update_command
     'update'
   end
 
+  def get_wanted_version(should)
+    if should =~ /[<>=~]/
+      begin
+        wanted_version = Puppet::Util::Package::Version::Range.parse(should, Version::Rpm)
+      rescue Puppet::Util::Package::Version::Range::ValidationFailure, Puppet::Util::Package::Version::Range::Rpm::ValidationFailure
+        raise Puppet::Error, _("Cannot parse %{should} as a RPM version") % { should: should }
+      end
+      # sort available versions in descending order
+      available_versions = get_available_versions(@resource[:name]).sort.reverse
+      available_versions.each do | available_version |
+        # returning the first match will return the highest version that satisfy the condition in the manifest
+        return available_version.to_string if wanted_version.include?(available_version)
+      end
+        self.debug "No available version that matches the desired semantic version"
```

Comment:
For easier debugging, I would include available versions in this print. Also, no version was found for the given range, not `semantic version`.